### ISSUE TITLE
🐛 [BUG] Different results when using model.train() and model.eval()

### ISSUE THREAD

**Issue by emilannevelink** [mir-group/nequip#224]
**Describe the bug**
I have a trained nequip model and try to evaluate it but am getting a value quite different than I expected. After some debugging, I realized that I wasn't setting the model to evaluate. So, i realized that if the model is in the training mode I get a different value that if it is in evaluation mode.

**To Reproduce**
Minimal code to reproduce the behavior. Please be try to isolate the code producing the error code from code specific to your task but not necessarily relevant to the error (e.g. replacing input data with random inputs instead of data from files). 

with a trained nequip model, 
model.train()(data) != model.eval()(data)

**Expected behavior**
I would expect the two to give the same result

**Environment (please complete the following information):**
 - OS: linux
 - python version Python 3.9.6
 - python environment (commands are given for python interpreter):
   - nequip version '0.5.3'
   - e3nn version '0.3.5'
   - pytorch version '1.9.0+cu102'
 - (if relevant) GPU support with CUDA
   -  cuda Version according to nvcc (`nvcc --version`)
   -  cuda version according to PyTorch '10.2'

**Additional context**

**Comment by Linux-cpp-lisp** [comment]
Hi @emilannevelink ,

In `train` mode, the model is not rescaling its predictions into "real" (eV or kcal/mol etc.) units. The `RescaleEnergyEtc` block is disabled in training mode. This is because training in unscaled ("normalized") units is likely more numerically stable, among other reasons.
